// @syncfusion/ej2-richtexteditor/src/markdown-parser/plugin/table.js
import * as CONSTANT from './../base/constant';
/**
 * Link internal component
 * @hidden
 */
var MDTable = /** @class */ (function () {
    /**
     * Constructor for creating the Formats plugin
     * @hidden
     */
    function MDTable(parent) {
        this.parent = parent;
        this.selection = this.parent.markdownSelection;
        this.addEventListener();
    }
    MDTable.prototype.addEventListener = function () {
        this.parent.observer.on(CONSTANT.MD_TABLE, this.createTable, this);
    };
    MDTable.prototype.removeEventListener = function () {
        this.parent.observer.off(CONSTANT.MD_TABLE, this.createTable);
    };
    MDTable.prototype.destroy = function () {
        this.removeEventListener();
    };
    MDTable.prototype.createTable = function (e) {
        var dummy = document;
        var textArea = this.parent.element;
        textArea.focus();
        var start = textArea.selectionStart;
        var end = textArea.selectionEnd;
        var end3;
        var text = this.selection.getSelectedText(textArea);
        var end1;
        var textEmpty;
        var textAreaInitial;
        if (start !== end) { // It will check start !== end and will clear the text selected.
            textEmpty = text.replace(text, '');
            end1 = end;
            end3 = start;
            text = textEmpty;
        }
        else {
            end3 = end;
        }
        text += this.textUnEmpty(start, end3, dummy, text, end1, textArea);
        for (var i = 1; i <= 2; i++) {
            text += '|';
            for (var j = 1; j <= 2; j++) {
                if (i === 1) {
                    text += e.item.headingText + ' ' + j + '|';
                }
                else {
                    text += '---------|';
                }
            }
            var dummyElement = dummy.createElement('div');
            dummyElement.innerHTML = '\n';
            var text1 = dummyElement.textContent;
            text += text1;
        }
        for (var i = 1; i <= 2; i++) {
            text += '|';
            for (var j = 1; j <= 2; j++) {
                text += e.item.colText + ' ' + this.convertToLetters(i) + j + '|';
            }
            var dummyElement = dummy.createElement('div');
            dummyElement.innerHTML = '\n';
            var text1 = dummyElement.textContent;
            text += text1;
        }
        text = this.textUnEmpty(start, end3, dummy, text, end1, textArea);
        textAreaInitial = textArea.value;
        if (start !== end) {
            this.startNotEqualEnd(start, end, text, textArea, textAreaInitial, e);
        }
        else if (start === 0 && end === 0) {
            textArea.value = textArea.value.substr(0, start)
                + text + textArea.value.substr(end, textArea.value.length);
            if (textAreaInitial.length) {
                this.restore(textArea, start + 3, end + 12, e);
            }
            else {
                this.restore(textArea, start + 1, end + 10, e);
            }
        }
        else {
            this.startEqualEnd(start, end, text, textArea, textAreaInitial, e);
        }
    };
    MDTable.prototype.startEqualEnd = function (start, end, text, textArea, textAreaInitial, e) {
        var parentText = this.selection.getSelectedParentPoints(textArea);
        var selectedLine = parentText.length - 1;
        var formatSplit;
        formatSplit = parentText[selectedLine].text.split(' ', 2);
        var textApplyFormat;
        var parentTextLength;
        if (formatSplit.length > 1) {
            parentTextLength = formatSplit[0].length + formatSplit[1].length + 1;
        }
        textApplyFormat = textArea.value.substring(end, textArea.value.length);
        if (start === parentTextLength) {
            textArea.value = textArea.value.substr(0, start)
                + text + textArea.value.substr(end, textArea.value.length);
            this.callRestore(textArea, start, end, e, textAreaInitial);
        }
        else if (textArea.value[start] === '2' && textArea.value[start + 1] === '.') {
            text = '';
            textArea.value = textArea.value.substr(0, start)
                + text + textArea.value.substr(end, textArea.value.length);
        }
        else if (!(textArea.value[start] === '#' || textArea.value[start - 1] === '#' ||
            textArea.value[start - 2] === '#' || textArea.value[start] === '2.' ||
            textArea.value[start - 1] === '2.' || textArea.value[start - 2] === '2.' || (textArea.value[start - 1] === '2' &&
            textArea.value[start] === '.') || (textArea.value[start - 1] === '.' && textArea.value[start] === ' ') ||
            (textArea.value[start - 1] === ' ' &&
                textArea.value[start - 2] === '.' && textArea.value[start - 3] === '2') ||
            textArea.value[start] === '>' || textArea.value[start - 1] === '>' || textArea.value[start - 2] === '>' ||
            textArea.value[start] === '+' || textArea.value[start - 1] === '+' || textArea.value[start - 2] === '+')) {
            if (!(parentText[0].text.match('#') || parentText[0].text.match('>') ||
                parentText[0].text.match('2.'))) {
                formatSplit[0] = '';
            }
            text += textApplyFormat.replace(textApplyFormat, (formatSplit[0] + ' ' + textApplyFormat));
            textArea.value = textArea.value.substr(0, start)
                + text;
            this.callRestore(textArea, start, end, e, textAreaInitial);
        }
        else {
            text = '';
            textArea.value = textArea.value.substr(0, start)
                + text + textArea.value.substr(end, textArea.value.length);
        }
    };
    MDTable.prototype.startNotEqualEnd = function (start, end, text, textArea, textAreaInitial, e) {
        var parentText = this.selection.getSelectedParentPoints(textArea);
        var textApplyFormat;
        textApplyFormat = textArea.value.substring(end, textArea.value.length);
        if (parentText.length < 2) {
            this.singleLine(start, end, text, textArea, parentText, textApplyFormat, textAreaInitial, e);
        }
        else {
            this.multipleLines(start, end, text, textArea, parentText, textApplyFormat, textAreaInitial, e);
        }
    };
    MDTable.prototype.singleLine = function (start, end, text, textArea, parentText, textApplyFormat, textAreaInitial, e) {
        var formatSplit;
        formatSplit = parentText[0].text.split(' ', 2);
        var selectedText;
        selectedText = this.selection.getSelectedText(textArea);
        var selectedTextSplit;
        selectedTextSplit = selectedText.split(' ', 2);
        if (selectedTextSplit.length === 2) {
            this.selectedSplitText(start, end, text, textArea, selectedText, parentText, formatSplit, textApplyFormat, e, textAreaInitial, selectedTextSplit);
        }
        else {
            if (textArea.value[start - 1] === ' ' && (textArea.value[start - 2] === '.' || textArea.value[start - 2] === '#' ||
                textArea.value[start - 2] === '>' || textArea.value[start - 2] === '+')) {
                text = '';
                start += selectedText.length;
                textArea.value = textArea.value.substr(0, start)
                    + text + textArea.value.substr(end, textArea.value.length);
            }
            else if (textArea.value[start] === '>' || textArea.value[start] === '+' || (textArea.value[start] === '2' &&
                textArea.value[start + 1] === '.') || (textArea.value[start] === '#' && textArea.value[start - 1] !== '#')) {
                if (textArea.value[end - 2] === '>' || textArea.value[end - 1] === '+' || textArea.value[end - 1] === '2' ||
                    textArea.value[end - 1] === '#' || (textArea.value[end - 1] === '.' && textArea.value[end - 2] === '2') ||
                    (textArea.value[end - 1] === ' ' && (textArea.value[end - 2] === '.' || textArea.value[end - 2] === '#' ||
                        textArea.value[end - 2] === '>' || textArea.value[end - 2] === '+'))) {
                    text = '';
                    start += selectedText.length;
                    textArea.value = textArea.value.substr(0, start)
                        + text + textArea.value.substr(end, textArea.value.length);
                }
                else {
                    if (!(parentText[0].text.match('#') || parentText[0].text.match('>') ||
                        parentText[0].text.match('2.'))) {
                        formatSplit[0] = '';
                    }
                    text += textApplyFormat.replace(textApplyFormat, (formatSplit[0] + ' ' + textApplyFormat));
                    textArea.value = textArea.value.substr(0, start)
                        + text;
                    this.callRestore(textArea, start, end, e, textAreaInitial);
                }
            }
            else {
                if (end === formatSplit[0].length + formatSplit[1].length + 1) {
                    textArea.value = textArea.value.substr(0, start)
                        + text + textArea.value.substr(end, textArea.value.length);
                    this.callRestore(textArea, start, end, e, textAreaInitial);
                }
                else {
                    if (!(parentText[0].text.match('#') || parentText[0].text.match('>') ||
                        parentText[0].text.match('2.'))) {
                        formatSplit[0] = '';
                    }
                    text += textApplyFormat.replace(textApplyFormat, (formatSplit[0] + ' ' + textApplyFormat));
                    textArea.value = textArea.value.substr(0, start)
                        + text;
                    this.callRestore(textArea, start, end, e, textAreaInitial);
                }
            }
        }
    };
    MDTable.prototype.selectedSplitText = function (start, end, text, textArea, selectedText, parentText, formatSplit, textApplyFormat, e, textAreaInitial, selectedTextSplit) {
        if (selectedTextSplit[0] === '') {
            if (textArea.value[start - 1] === '#' || textArea.value[start - 1] === '.' ||
                textArea.value[start - 1] === '>' || textArea.value[start - 1] === '+') {
                text = '';
                start += selectedText.length;
                textArea.value = textArea.value.substr(0, start)
                    + text + textArea.value.substr(end, textArea.value.length);
            }
            else {
                if (!(parentText[0].text.match('#') || parentText[0].text.match('>') ||
                    parentText[0].text.match('2.'))) {
                    formatSplit[0] = '';
                }
                text += textApplyFormat.replace(textApplyFormat, (formatSplit[0] + ' ' + textApplyFormat));
                textArea.value = textArea.value.substr(0, start)
                    + text;
                this.callRestore(textArea, start, end, e, textAreaInitial);
            }
        }
        else {
            if (textArea.value[start] === '>' || textArea.value[start] === '+' || textArea.value[start] === '#' ||
                textArea.value[start] === '2' || (textArea.value[start] === '.' && textArea.value[start - 1] === '2')) {
                if (selectedText.length === (formatSplit[0].length + formatSplit[1].length + 1)) {
                    textArea.value = textArea.value.substr(0, start)
                        + text + textArea.value.substr(end, textArea.value.length);
                }
                else if (textArea.value[start] === '>' || textArea.value[start] === '+' || (textArea.value[start] === '2' &&
                    textArea.value[start + 1] === '.') || (textArea.value[start] === '#' && textArea.value[start - 1] !== '#')) {
                    if (!(textArea.value[end - 2] === '>' || textArea.value[end - 1] === '+' || textArea.value[end - 1] === '#' ||
                        (textArea.value[end - 1] === '.' && textArea.value[end - 2] === '2') || (textArea.value[end - 1] === ' ' &&
                        (textArea.value[end - 2] === '.' || textArea.value[end - 2] === '#' || textArea.value[end - 2] === '>' ||
                            textArea.value[end - 2] === '+')))) {
                        if (!(parentText[0].text.match('#') || parentText[0].text.match('>') ||
                            parentText[0].text.match('2.'))) {
                            formatSplit[0] = '';
                        }
                        text += textApplyFormat.replace(textApplyFormat, (formatSplit[0] + ' ' + textApplyFormat));
                        textArea.value = textArea.value.substr(0, start)
                            + text;
                        this.callRestore(textArea, start, end, e, textAreaInitial);
                    }
                    else {
                        text = '';
                        start += selectedText.length;
                        textArea.value = textArea.value.substr(0, start)
                            + text + textArea.value.substr(end, textArea.value.length);
                    }
                }
                else {
                    text = '';
                    start += selectedText.length;
                    textArea.value = textArea.value.substr(0, start)
                        + text + textArea.value.substr(end, textArea.value.length);
                }
            }
        }
    };
    MDTable.prototype.multipleLines = function (start, end, text, textArea, parentText, textApplyFormat, textAreaInitial, e) {
        var lastSelectedLineIndex = parentText.length - 1;
        var formatLastLine;
        formatLastLine = parentText[lastSelectedLineIndex].text.split(' ', 2);
        var formatFirstLine;
        formatFirstLine = parentText[0].text.split(' ', 2);
        var selectedText;
        selectedText = this.selection.getSelectedText(textArea);
        if (textArea.value[start - 1] === '#' || (textArea.value[start - 1] === '.' && textArea.value[start - 2] === '2') ||
            textArea.value[start - 1] === '>' || textArea.value[start - 1] === '+' ||
            textArea.value[start - 1] === '2' || (textArea.value[start - 1] === ' ' &&
            (textArea.value[start - 2] === '#' || textArea.value[start - 2] === '>' ||
                textArea.value[start - 2] === '+' || textArea.value[start - 2] === '.'))) {
            text = '';
            start += selectedText.length;
            textArea.value = textArea.value.substr(0, start)
                + text + textArea.value.substr(end, textArea.value.length);
        }
        else if (textArea.value[end] === '#' || textArea.value[end] === '>' || textArea.value[end] === '+' ||
            textArea.value[end] === '2' || (textArea.value[end] === '.' && textArea.value[end - 1] === '2') ||
            (textArea.value[end - 1] === ' ' && (textArea.value[end - 2] === '.' || textArea.value[end - 2] === '#' ||
                textArea.value[end - 2] === '>' || textArea.value[end - 2] === '+')) || (textArea.value[end] === ' ' &&
            (textArea.value[end - 1] === '#' || textArea.value[end - 1] === '>' || textArea.value[end - 1] === '+' ||
                textArea.value[end - 1] === '.'))) {
            text = '';
            start += selectedText.length;
            textArea.value = textArea.value.substr(0, start)
                + text + textArea.value.substr(end, textArea.value.length);
        }
        else {
            if (!(parentText[lastSelectedLineIndex].text.match('#') ||
                parentText[lastSelectedLineIndex].text.match('>') ||
                parentText[lastSelectedLineIndex].text.match('2.'))) {
                formatLastLine[0] = '';
            }
            text += textApplyFormat.replace(textApplyFormat, (formatLastLine[0] + ' ' + textApplyFormat));
            textArea.value = textArea.value.substr(0, start)
                + text;
            this.callRestore(textArea, start, end, e, textAreaInitial);
        }
    };
    MDTable.prototype.convertToLetters = function (rowNumber) {
        var baseChar = ('A').charCodeAt(0);
        var letters = '';
        do {
            rowNumber -= 1;
            letters = String.fromCharCode(baseChar + (rowNumber % 26)) + letters;
            rowNumber = (rowNumber / 26) >> 0;
        } while (rowNumber > 0);
        return letters;
    };
    MDTable.prototype.textUnEmpty = function (start, end, dummy, text, end1, textArea) {
        if (start === end && ((start !== 0 && end !== 0) || end1 !== 0)) {
            var dummyElement = dummy.createElement('div');
            if (!(text.length > 0)) {
                if (textArea.value.length > 0) {
                    dummyElement.innerHTML = '\n\n';
                }
                else {
                    dummyElement.innerHTML = '';
                }
            }
            else {
                dummyElement.innerHTML = '\n';
            }
            var text1 = dummyElement.textContent;
            return text += text1;
        }
        else {
            return text;
        }
    };
    MDTable.prototype.callRestore = function (textArea, start, end, e, textAreaInitial) {
        if (textAreaInitial.length) {
            this.restore(textArea, start + 3, start + 12, e);
        }
        else {
            this.restore(textArea, start + 1, end + 10, e);
        }
    };
    MDTable.prototype.restore = function (textArea, start, end, event) {
        this.selection.save(start, end);
        this.selection.restore(textArea);
        if (event && event.callBack) {
            event.callBack({
                requestType: event.subCommand,
                selectedText: this.selection.getSelectedText(textArea),
                editorMode: 'Markdown',
                event: event.event
            });
        }
    };
    return MDTable;
}());
export { MDTable };
