// @syncfusion/ej2-angular-dropdowns/src/multi-select/multiselect.component.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, ElementRef, ViewContainerRef, Renderer2, Injector, ChangeDetectionStrategy, forwardRef, ContentChild } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
import { ComponentBase, ComponentMixins, FormBase, setValue } from '@syncfusion/ej2-angular-base';
import { MultiSelect } from '@syncfusion/ej2-dropdowns';
import { Template } from '@syncfusion/ej2-angular-base';
export var inputs = ['actionFailureTemplate', 'allowCustomValue', 'allowFiltering', 'closePopupOnSelect', 'cssClass', 'dataSource', 'delimiterChar', 'enablePersistence', 'enableRtl', 'enableSelectionOrder', 'enabled', 'fields', 'filterBarPlaceholder', 'floatLabelType', 'footerTemplate', 'groupTemplate', 'headerTemplate', 'hideSelectedItem', 'htmlAttributes', 'ignoreAccent', 'ignoreCase', 'itemTemplate', 'locale', 'maximumSelectionLength', 'mode', 'noRecordsTemplate', 'openOnClick', 'placeholder', 'popupHeight', 'popupWidth', 'query', 'readonly', 'selectAllText', 'showClearButton', 'showDropDownIcon', 'showSelectAll', 'sortOrder', 'text', 'unSelectAllText', 'value', 'valueTemplate', 'width', 'zIndex'];
export var outputs = ['actionBegin', 'actionComplete', 'actionFailure', 'beforeOpen', 'blur', 'change', 'chipSelection', 'close', 'created', 'customValueSelection', 'dataBound', 'destroyed', 'filtering', 'focus', 'open', 'removed', 'removing', 'select', 'selectedAll', 'tagging', 'valueChange'];
export var twoWays = ['value'];
/**
* The MultiSelect allows the user to pick a values from the predefined list of values.
*```html
*<ejs-multiselect></ejs-multiselect>
*```
*/
var MultiSelectComponent = /** @class */ (function (_super) {
    __extends(MultiSelectComponent, _super);
    function MultiSelectComponent(ngEle, srenderer, viewContainerRef, injector) {
        var _this = _super.call(this) || this;
        _this.ngEle = ngEle;
        _this.srenderer = srenderer;
        _this.viewContainerRef = viewContainerRef;
        _this.injector = injector;
        _this.skipFromEvent = true;
        _this.element = _this.ngEle.nativeElement;
        _this.injectedModules = _this.injectedModules || [];
        try {
            var mod = _this.injector.get('DropDownsCheckBoxSelection');
            if (_this.injectedModules.indexOf(mod) === -1) {
                _this.injectedModules.push(mod);
            }
        }
        catch (_a) { }
        _this.registerEvents(outputs);
        _this.addTwoWay.call(_this, twoWays);
        setValue('currentInstance', _this, _this.viewContainerRef);
        return _this;
    }
    MultiSelectComponent_1 = MultiSelectComponent;
    MultiSelectComponent.prototype.registerOnChange = function (registerFunction) {
    };
    MultiSelectComponent.prototype.registerOnTouched = function (registerFunction) {
    };
    MultiSelectComponent.prototype.writeValue = function (value) {
    };
    MultiSelectComponent.prototype.setDisabledState = function (disabled) {
    };
    MultiSelectComponent.prototype.ngOnInit = function () {
    };
    MultiSelectComponent.prototype.ngAfterViewInit = function () {
    };
    MultiSelectComponent.prototype.ngOnDestroy = function () {
    };
    MultiSelectComponent.prototype.ngAfterContentChecked = function () {
    };
    MultiSelectComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ejs-multiselect',
                    inputs: inputs,
                    outputs: outputs,
                    template: '',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [
                        {
                            provide: NG_VALUE_ACCESSOR,
                            useExisting: forwardRef(function () { return MultiSelectComponent_1; }),
                            multi: true
                        }
                    ],
                    queries: {}
                },] },
    ];
    /** @nocollapse */
    MultiSelectComponent.ctorParameters = function () { return [
        { type: ElementRef, },
        { type: Renderer2, },
        { type: ViewContainerRef, },
        { type: Injector, },
    ]; };
    MultiSelectComponent.propDecorators = {
        'footerTemplate': [{ type: ContentChild, args: ['footerTemplate',] },],
        'headerTemplate': [{ type: ContentChild, args: ['headerTemplate',] },],
        'valueTemplate': [{ type: ContentChild, args: ['valueTemplate',] },],
        'itemTemplate': [{ type: ContentChild, args: ['itemTemplate',] },],
        'noRecordsTemplate': [{ type: ContentChild, args: ['noRecordsTemplate',] },],
        'actionFailureTemplate': [{ type: ContentChild, args: ['actionFailureTemplate',] },],
    };
    __decorate([
        Template(),
        __metadata("design:type", Object)
    ], MultiSelectComponent.prototype, "footerTemplate", void 0);
    __decorate([
        Template(),
        __metadata("design:type", Object)
    ], MultiSelectComponent.prototype, "headerTemplate", void 0);
    __decorate([
        Template(),
        __metadata("design:type", Object)
    ], MultiSelectComponent.prototype, "valueTemplate", void 0);
    __decorate([
        Template(),
        __metadata("design:type", Object)
    ], MultiSelectComponent.prototype, "itemTemplate", void 0);
    __decorate([
        Template('No Records Found'),
        __metadata("design:type", Object)
    ], MultiSelectComponent.prototype, "noRecordsTemplate", void 0);
    __decorate([
        Template('The Request Failed'),
        __metadata("design:type", Object)
    ], MultiSelectComponent.prototype, "actionFailureTemplate", void 0);
    MultiSelectComponent = MultiSelectComponent_1 = __decorate([
        ComponentMixins([ComponentBase, FormBase]),
        __metadata("design:paramtypes", [ElementRef, Renderer2, ViewContainerRef, Injector])
    ], MultiSelectComponent);
    return MultiSelectComponent;
    var MultiSelectComponent_1;
}(MultiSelect));
export { MultiSelectComponent };
